src/filePush/filePush.ts

```diff
@@ -11,40 +11,42 @@ export interface Config {
   privateKey?: string;
 }
 
-export const createConfig = async () => {
-  const config: Config = {
-    host: process.env.SFTP_Address,
-    username: process.env.SFTP_User,
-    retries: 3,
-  };
+type Feeds = 'evl' | 'tfl';
+
+export const createConfig = async (eventType: string) => {
+  let secretString: string;
 
-  if (process.env.SFTP_Key && process.env.SFTP_Key != '') {
-    const sftpKey = await getSecret(process.env.SFTP_Key);
-    config.privateKey = sftpKey;
-  } else if (process.env.SFTP_Password && process.env.SFTP_Password != '') {
-    const sftpPassword = await getSecret(process.env.SFTP_Password);
-    config.password = sftpPassword;
+  if (eventType === 'evl') {
+    secretString = await getSecret(process.env.EVL_SFTP_CONFIG);
+  } else if (eventType === 'tfl') {
+    secretString = await getSecret(process.env.TFL_SFTP_CONFIG);
   } else {
-    logger.error(
-      '',
-      'No password or private key found, please check the env variables',
-    );
-    throw new Error(
-      'No password or private key found, please check the env variables',
-    );
+    logger.error('', 'Unable to determine event type, please try again');
   }
 
-  return config;
+  try {
+    return JSON.parse(secretString) as Config;
+  } catch (e) {
+    logger.error('', e);
+    throw e;
+  }
 };
 
-export const filePush = async (filepath: string) => {
-  const config = await createConfig();
+export const filePush = async (filepath: string, eventType: string) => {
+  const config = await createConfig(eventType);
+  logger.info('Created config from secrets');
   const sftp = new Client();
+  const sftpPath: { [key in Feeds]: string } = {
+    evl: process.env.EVL_SFTP_PATH,
+    tfl: process.env.TFL_SFTP_PATH,
+  };
   const remoteFileLocation =
-    (process.env.SFTP_Path ?? '') + path.basename(filepath);
+    ((sftpPath[eventType] as string) ?? '') + path.basename(filepath);
```

Comment:
Can avoid this warning by doing this I think... `sftPath[eventType as string]`

Reply:
Nah just tells me no need to do that because it's an uneeded assertion, the only way I think I can remove this is to just disable it. It isn't what it's warning of anyway

Reply:
😢